---
title: "Mendelian randomization for causal inference in Genome-wide association studies"
author: "Frederick J. Boehm & Xiang Zhou"
date: "`r lubridate::now()`"
bibliography: exported.bib
csl: elsevier-with-titles.csl
output: 
  - bookdown::html_document2
  - bookdown::word_document2
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




# Introduction

Great interest has developed in inferring causal relationships between complex 
traits, *i.e.*, traits that seemingly are not inherited in a Mendelian fashion, in observational human genetics studies. Discovery of such relationships is crucial to 
enhancing our understanding of the biology of health and disease. Two-sample 
Mendelian randomization (MR) studies are a form of instrumental variable analysis in which
genetic markers, typically single nucleotide polymorphisms (SNPs), serve as instruments for 
inferring causal effects of an exposure variable on an outcome variable [@wright1928tariff;@wright1921correlation;@wright1923theory;@stock2003retrospectives;@katan1986apolipoprotein;@katan2004commentary;@didelez2007mendelian]. They are termed 
"two-sample" MR studies because the exposure variable and the outcome variable are measured on 
two non-overlapping cohorts of subjects. Importantly, the SNP instruments are measured in both sets of subjects.

Recent developments have enabled performance of two-sample MR studies with only 
GWAS summary statistics [@burgess2015using].


Genome-wide association studies (GWAS) present unique opportunities for 
discovery of causal relationships via MR. A GWAS
interrogates millions of single nucleotide polymorphisms (SNPs) to infer which affect the trait. A 
SNP is a genetic marker defined by nucleotide base identity at a single genomic position.

```{r, article-counts, fig.cap= "Upward trend in article counts by year for keyword searches: 1. mendelian randomization and 2. genome-wide association study."}
knitr::include_graphics("figures/mr-gwas-counts-scatter.png")
```


Methods for Mendelian randomization have appeared at a rapid and accelerating pace since the 
appearance of the first GWAS (Figure \@ref(fig:article-counts)). 

Now, GWAS and Mendelian randomization for biobank-scale data offer the possibility
of discovering even more causal relationships among complex traits. The large
sample sizes in biobank studies offer the possibility of uncovering associations of modest strength 
that might be undetectable in smaller sample sizes. 

~~Investigating causal relationship among complex traits and identifying causal risk
factors are an important first step towards understanding the biology of diseases. A
common statistical tool for performing such causal inference in observational studies is
Mendelian randomization (MR). MR is a form of instrumental variable analysis that
uses SNPs to serve as instruments for inferring the causal effect of an exposure variable
on an outcome variable(1). MR requires only summary statistics from genome-wide
association studies (GWASs) and is often performed in a two-sample study setting
where the exposure variable and the outcome variable are measured in two separate
studies(2). With the abundant availability of GWAS summary statistics, numerous MR
analyses are being carried out, identifying important causal risk factors for various
common diseases. These MR studies are facilitated by many recently developed MR
methods that include the inverse variance weighted (IVW) method, MR-Egger(3),
median-based regression(4), BWMR(5), RAPS(6), MRMix(7), CAUSE(8), to name a
few. Different MR methods differ in their modeling assumptions and inference
algorithms, but the majority of them encounter two important modeling and algorithmic
challenges that have so far limited the effectiveness of MR analysis.
First, almost all existing MR methods rely on a pre-selected set of independent
SNPs to serve as instruments for MR analysis. The instruments are selected to be
independent from each other to ensure the validity of the statistical inference framework
used in many common MR methods such as IVW. The independent SNPs are often
selected through linkage disequilibrium (LD) clumping, a procedure that first ranks
SNPs based on their marginal association evidence with the exposure variable and then
retains SNPs that are not in high LD with the SNPs on top of the ranking list. Using LD
clumping to select SNPs may be suboptimal, however, as the selected SNPs may only
epresent tagging SNPs that are in LD with the causal SNPs rather than the causal ones
themselves. Using tagging SNPs instead of the causal ones to serve as instruments can
reduce the power of MR analysis. In addition, perhaps more importantly, selecting
independent SNPs for MR analysis may not be ideal either, as complex traits can be
influenced by multiple causal SNPs residing in the same local region that are in
potential LD with each other. Consequently, selecting independent SNPs may only
capture a small proportion of the phenotypic variance in the exposure variable, again
leading to a loss of power in the subsequent MR analysis (1, 2, 9, 10). Indeed, in the
parallel research field of transcriptome-wide association studies, it has been well
documented that incorporating correlated SNPs can substantially improve analysis
power than using independent SNPs only(11-14). Therefore, incorporating correlated
SNPs and developing effective approaches to select instruments among them are
important to fully captivate the potential of MR.
Second, only a limited number of MR methods model horizontal pleiotropy and
even fewer can effectively control for it during MR analysis(15). Horizontal pleiotropy
occurs when the SNP instruments exhibit effects on the outcome through pathways
other than the exposure. Horizontal pleiotropy has been widely observed in complex
trait analysis(13, 15) and often comes in two distinct types. The first type of horizontal
pleiotropy arises through paths independent of the exposure, with the resulting
horizontal pleotropic effects being independent of the SNP effects on the exposure. The
second type of horizontal pleiotropy arises through unobserved exposure-outcome
confounders and induces correlation between the horizontal pleotropic effects and the
SNP effects on the exposure. The presence of either type of horizontal pleiotropy
violates standard MR modeling assumptions and can lead to biased causal effect
estimates and increased false discoveries. Early MR analyses control for horizontal~~






# Mendelian Randomization with one Instrument

We begin our discussion of Mendelian randomization by considering the simplest case, where there is a single genetic marker and a single outcome variable. Mendelian randomization with one instrument uses a single genetic marker 
as the proxy ("instrument" or "instrumental variable") for the "exposure", a complex trait, and asks 
whether variation in the genotypes at the genetic marker causes variation in the 
"outcome" variable, which is typically a second complex trait. When the genetic marker 
is a single nucleotide polymorphism on an autosome, the cohort is partitioned into three 
classes, two homozygote classes and a single heterozygote class, based on genotypes 
for that marker. When the three assumptions of mendelian randomization are satisfied, 
estimation of the causal effect of the exposure on the outcome is possible. 

The outcome variable can be binary, a count, or a continuous variable. 

## Assumptions of Mendelian Randomization

Three assumptions are central to Mendelian Randomization. Satisfying them is necessary to infer 
causal effects in a MR study. Careful evaluation of the assumptions is needed in MR studies. Briefly, 
the first assumption states that the exposure is associated with the outcome. The second requires 
independence, conditional on the exposure and all measured and unmeasured confounders, between the 
genetic marker and the outcome. The last of the three assumptions states that the genetic marker is 
independent of all (measured and unmeasured) confounders of the relationship between exposure and 
outcome. We next discuss these in more detail and consider their evaluation in MR studies.

If the first assumption, about the association between the genetic marker and the exposure, holds, 
but this association is weak, then an amplification of biases, such as those due to violations of 
assumptions 2 and 3, may result [@vanderweele_methodological_2014]. 

The second assumption, often termed the exclusion restriction, can also be stated as 
the need for the genetic marker to affect the outcome only through the exposure. In a 
causal diagram, this means that the only path from the marker to the outcome is that 
containing the exposure.

The third assumption, sometimes called the ____, ensures that ___. 

When the three MR assumptions hold, one may use instrumental variable statistical methods, 
with the genetic marker data, the exposure data, and the outcome data, to estimate and to test the 
causal effect of the exposure on the outcome. 


## Statistical models for MR with one instrument and one outcome



## Statistical methods for MR with one instrument and one outcome

The approaches to MR can be classified into two categories: two-stage methods and 
likelihood-based methods. Both two-stage strategies and likelihood-based strategies 
make parametric assumptions for the identification of causal effects [@burgess_review_2017].


Two-stage methods use two regressions. In the first stage, the exposure is regressed on
the instrument to obtain fitted values. In the second stage, the outcome is regressed on the fitted values for the 
exposure, which result from the first regression. Note that the uncertainty in the fitted values of 
the first regression is not considered when performing the second regression.


## Assumption violations

There are several ways that real data may violate the three MR assumptions. These include . 

Recognition that the assumptions may be violated has inspired methods advances that 
enable assumption relaxations under certain conditions. 

## Sensitivity analysis for MR with one instrument and one outcome

Sensitivity analysis is recommended and widely used in MR studies because of the inability to verify 
the three MR assumptions with observational data. The goal of a sensitivity analysis is to gain insight 
into how the results might differ were we able to verify the MR assumptions.










# Mendelian Randomization with multiple instruments

Recognition of complex traits' diverse genetic architectures has fueled development of MR methods with multiple instruments [@Boyle2017]. 




# Horizontal pleiotropy

Horizontal pleiotropy refers to the presence of multiple pathways from a genetic 
variant to the outcome. One of these multiple pathways may contain the exposure variable.
The fact that there are multiple pathways from the genetic variant to the outcome violates
the MR assumptions. @verbanck_detection_2018 documented widespread horizontal pleiotropy 
in GWAS. This finding motivated the development of MR-PRESSO and inspired 
further research into methods to accommodate horizontal pleiotropy in MR studies [@verbanck_detection_2018].
* What does MR-PRESSO do? What are its limitations?







# Correlated horizontal pleiotropy

@yuan_testing_2020 


# Recent Findings

## Omnigenic MR

Recent research on the omnigenic hypothesis, which posits 
that every SNP's effect on a trait is nonzero, has informed
MR methods development [@Boyle2017]. @wang_mendelian_nodate 
recognized the limitations of previous MR methods in the 
context of the omnigenic hypothesis. To accommodate the possibility
that a trait is influenced by many genome-wide, small effects, 

## Multiple outcomes in MR

Investigators have found that combining information from multiple related traits can lead to richer MR results. 

One approach to combining information across traits is to extend the univariate MR statistical methods to their multivariate analogs.

# Applications of MR in GWAS 














# Summary and Outlook





# References

